### PR TITLE
Linux 5.4.173 patch release 6/30/22

### PR DESCRIPTION
**Pull request by dac2**

### CODE REVIEW

**Comment by jbradleysmith** [comment]
Rob told me he had to apply this patch to his 5.4 kernel:
```
commit eeabad79ee605c65b6a670971a13d31182181f87
Author: Rob Gardner <rob.gardner@oracle.com>
Date:   Fri May 13 14:40:30 2022 -0600

    Hook to platform_serror
    
    Give the platform an opportunity to examine the
    circumstances around an serror before going into
    a panic.
    
    This is necessary for some pensando related stuff.
    I'll write something more convincing here.
    
    Signed-off-by: Rob Gardner <rob.gardner@oracle.com>

diff --git a/arch/arm64/kernel/traps.c b/arch/arm64/kernel/traps.c
index 4b0e02cb68e0..edad397afc8b 100644
--- a/arch/arm64/kernel/traps.c
+++ b/arch/arm64/kernel/traps.c
@@ -906,10 +906,18 @@ bool arm64_is_fatal_ras_serror(struct pt_regs *regs, unsigned int esr)
 	}
 }
 
+int __weak platform_serror(struct pt_regs *regs, unsigned int esr)
+{
+        return 0;
+}
+
 asmlinkage void do_serror(struct pt_regs *regs, unsigned int esr)
 {
 	const bool was_in_nmi = in_nmi();
 
+	if (platform_serror(regs, esr))
+		return;
+
 	if (!was_in_nmi)
 		nmi_enter();
 ```

**Comment by dac2** [comment]
@jbradleysmith Take a look at the new version.  This removes the hooking of bad_mode(), which we also don't hook in 5.10, and instead hooks do_serror().  The cap_pcie.c function name is platform_serror(), so it matches already.

File: dsc-linux-5.4.173/patches-2022-06/00_elba_serror_fix.patch [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+-		return;
+-
+ 	console_verbose();
+ 
+ 	pr_crit("Bad mode in %s handler detected on CPU%d, code 0x%08x -- %s\n",
+@@ -906,10 +898,18 @@ bool arm64_is_fatal_ras_serror(struct pt_regs *regs, unsigned int esr)
+ 	}
+ }
+ 
++int __weak platform_serror(struct pt_regs *regs, unsigned int esr)
```
Comment (jbradleysmith) [inline dsc-linux-5.4.173/patches-2022-06/00_elba_serror_fix.patch:44]: traps.h has a function prototype for `platform_bad_mode` that should also change to `platform_serror`.

**Comment by dac2** [comment]
Ack; pushed again...

**Review by jbradleysmith — APPROVED** [review]
lgtm!

**Pull request merged**